File: algorithm/src/string/l000383RansomNote.py

class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        freq = [0] * 26

        # 'a'=97
        for c in magazine:
            freq[ord(c) - 97] += 1
        
        for c in ransomNote:
            temp = ord(c) - 97
            freq[temp] -= 1

            if freq[temp] < 0 :
                return False

        return True

File: algorithm/src/string/l000383RansomNote.go
package main

//import "fmt"

func canConstruct(ransomNote string, magazine string) bool {
	var dict [26]int

	for i:=0; i < len(magazine); i++{
		dict[ int(rune(magazine[i]) - rune('a')) ]++
	}

	for j:=0; j < len(ransomNote); j++{
		dict[ int(rune(ransomNote[j]) - rune('a')) ] --

		if( dict[ int(rune(ransomNote[j]) - rune('a')) ] < 0 ){
			return false;
		}
		
	}

	return false;
}